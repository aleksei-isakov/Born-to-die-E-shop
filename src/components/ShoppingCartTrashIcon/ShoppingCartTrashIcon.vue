<template>
  <base-custom-icon
    class="trash-icon"
    :width="25"
    icon="delete"
    @click="onClickDeleteItem"
  />
</template>

<script>
import { BaseCustomIcon } from '@/base_components/';

export default {
  name: 'ShoppingCartTrashIcon',

  components: {
    BaseCustomIcon
  },

  methods: {
    onClickDeleteItem() {
      this.$emit('delete-item');
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.trash-icon {
  cursor: pointer;
  flex-shrink: 0;

  @media screen and (max-width: $mobile-size) {
    position: absolute;
    top: 50%;
    right: -45px;
    transform: translateY(-50%);
  }
}
</style>
