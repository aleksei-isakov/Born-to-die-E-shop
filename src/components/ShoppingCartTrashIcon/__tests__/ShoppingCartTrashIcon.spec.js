import { createLocalVue, shallowMount } from '@vue/test-utils';
import ShoppingCartTrashIcon from '@/components/ShoppingCartTrashIcon/ShoppingCartTrashIcon';
import Vuex from 'vuex';
import ShoppingCartModule from '@/store/modules/shoppingcart';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('ShoppingCartTrashIcon', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        ShoppingCartModule: {
          namespaced: true,
          state: ShoppingCartModule.state,
          getters: ShoppingCartModule.getters,
          actions: ShoppingCartModule.actions
        }
      }
    });
  });
  test('should import BaseCustomIcon', () => {
    wrapper = shallowMount(ShoppingCartTrashIcon, {
      localVue,
      store
    });
    expect(wrapper.find('BaseCustomIcon')).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
