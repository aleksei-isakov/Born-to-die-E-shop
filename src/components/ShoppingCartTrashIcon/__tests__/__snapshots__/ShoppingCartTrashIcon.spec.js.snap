// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCartTrashIcon should match snapshot 1`] = `
<div class="trash-icon__wrapper">
  <base-custom-icon-stub icon="delete" width="25" class="trash-icon"></base-custom-icon-stub>
</div>
`;
